### PR TITLE
docs: add longbridge whale case

### PR DESCRIPTION
**Pull request by milasuperstar**
🏦 case study

### CODE REVIEW

File: content/blog/longbridge-case-study.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: Case Study - Longbridge Whale
+author: Changyu
+published_at: 2023/06/09 21:21:21
+feature_image: /content/blog/longbridge-case-study/banner.webp
+tags: Industry
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:6]: Case Study

File: content/blog/longbridge-case-study.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Theoretically, tenant databases are required to be isomorphic, but in reality, due to the manual management of change scripts or random temporary emergencies, there will always be more or less schema differences for some databases. This can lead to the failure of unified changes being released in only some of the tenant databases. Imagine the time-consuming troubleshooting process.
+
+### Adding new tenants
+
+New tenants are usually created by the business side. For most SaaS companies, the team that manages the databases are not immediately notified, resulting in a time difference spanning from hours to days for incorporating the new tenant database. This time difference is enough to generate a large number of schema differences.
+
+## The Solution: Bytebase
+
+As a financial SaaS provider, Longbridge Whale manage their database in tenant mode. With the growth of their business, the number of databases increased rapidly and schema management issues have become prominent.
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:47]: Add a link to our tenant mode doc

File: content/blog/longbridge-case-study.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+![_](/content/blog/longbridge-case-study/drift-detection.webp)
+
+In production environments, it is still difficult to avoid drifts. For example, access permissions are not fully revoked, and one of the app developers makes changes through other clients; or in some emergency situations, certain scripts are released directly to the database. Once a drift occurs, it is often difficult to discover in time. When this drift triggers an error at some point in the future, it will not be easy to trace back and find the cause, making it that much harder to be fixed. Bytebase has automatic [Drift Detection](/docs/change-database/drift-detection/) that scans the database schema regularly so as to detect unaudited changes and issue alerts promptly.
+
+### Recover: Reconcile the Schema Diff
+
+Once a drift occurs, Sync Schema can quickly compare the differences and generate issues to reconcile the drift.
+
+## Automated New Tenant Incorporation
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:79]: Automated New Tenant Provision

File: content/blog/longbridge-case-study.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Another situation prone to schema drift is the supply of new tenant databases. Previously for Longbridge Whale, as the business side creates a new tenant (through the frontend), their platform automatically creates a tenant database, syncs it to match a certain version of the schema, and initializes data. Many SaaS companies have also adopted similar solutions.
+
+The problem that follows is that if the tech team uses a separate tool for change management, the new tenant database will not be automatically incorporated unless the team is notified or if they scan all databases on a regular basis. This gap is enough to produce new schema differences. Thus, it is necessary to integrate this entire supply process with management process to achieve full automation. Here, Bytebase offers two solutions:
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:83]: incorporated -> enlisted

File: content/blog/longbridge-case-study.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+![_](/content/blog/longbridge-case-study/api-solution.webp)
+
+The application will call on the Bytebase API to create a new database and automatically include it into change management. Bytebase will make sure the schema is synced, while the application will complete subsequent data initialization.
+
+**Terraform Solution**
+
+![_](/content/blog/longbridge-case-study/terraform-solution.webp)
+
+The application calls on the Terraform Provider of the database instance supplier to create a new database, and then calls the [Bytebase Terraform Provider](https://registry.terraform.io/providers/bytebase/bytebase/latest/docs) for automatic management. The rest is pretty similar: Bytebase syncs database schema and the application initialized data.
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:95]: initialized -> initializes

File: content/blog/longbridge-case-study.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+## Longbridge Whale's Multi Tenant Solution Based on Bytebase
+
+![_](/content/blog/longbridge-case-study/longbridge-whale-bytebase.webp)
+
+Longbridge Whale has migrated their entire database management process to Bytebase, from tenant database creation to the following management:
+
+1. Whenever a new tenant is to be created, their SCM calls Bytebase API to create a new tenant database.
+2. Bytebase automatically synchronizes the schema of the new tenant database to the latest version.
+3. The application obtains database access permissions based on Vault and initializes data.
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:107]: HashiCorp Vault

File: content/blog/longbridge-case-study.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+4. SRE and business development teams complete daily changes to tenant databases through Bytebase.
+
+At the moment, Longbridge Whale and Bytebase are working together to further incorporate Bytebase's capabilities into their development process. Stay tuned for the next case study.
```
Comment (tianzhou) [inline content/blog/longbridge-case-study.md:110]: Stay tuned for a followup case study

**Review by tianzhou — CHANGES_REQUESTED** [review]

File: content/blog/longbridge-case-study.md (hunk 3 above)
Comment (milasuperstar) [reply content/blog/longbridge-case-study.md:79]: ah yes provision. that's the word i was looking for

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**